packages/core/src/render3/error_code.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+export const enum RuntimeErrorCode {
+  // Internal Errors
+
+  // Change Detection Errors
+  EXPRESSION_CHANGED_AFTER_CHECKED = '100',
+
+  // Dependency Injection Errors
+  INSTANTIATE_CYCLIC_DEPENDENCY = '200',
+  PROVIDER_NOT_FOUND = '201',
+
+  // Template Errors
+  MULTIPLE_COMPONENTS_MATCH = '300',
+  EXPORT_NOT_FOUND = '301',
+  PIPE_NOT_FOUND = '302',
+  BINDING_NOT_KNOWN = '303',
+  NOT_KNOWN_ELEMENT = '304',
+
+  // Styling Errors
+
+  // Declarations Errors
+
+  // i18n Errors
+
+  // Compilation Errors
+}
+
+export class RuntimeError extends Error {
+  constructor(public code: RuntimeErrorCode, message: string) {
+    super(`${formatRuntimeErrorCode(code)}${message}`);
+  }
+}
+
+/** Called to format a runtime error code */
+export function formatRuntimeErrorCode(code?: RuntimeErrorCode): string {
```

Comment:
Can we have the `code` argument non-optional?
```suggestion
export function formatRuntimeErrorCode(code: RuntimeErrorCode): string {
```
It looks like this argument is always defined when the function is called.

Reply:
I think we can extend this function to format the entire message, since it's used for `console.error` calls too.
The function may look like this:
```typescript
export function formatRuntimeError(code: RuntimeErrorCode, message: string): string {
  const fullCode = code ? `NG0${code}: ` : '';
  return `${fullCode}${message}`;
}
```
So in the `RuntimeError` class the logic will look like this:
```typescript
 super(formatRuntimeError(code, message);
```
and similarly for the `console.error` calls:
```typescript
function logUnknownPropertyError(propName: string, tNode: TNode): void {
  const message = `Can't bind to '${propName}' since it isn't a known property of '${tNode.value}'.`;
  console.error(formatRuntimeError(RuntimeErrorCode.BINDING_NOT_KNOWN, message));
}
```
What do you think?